### ISSUE TITLE
PyTorch中在反向传播前为什么要手动将梯度清零？

### ISSUE THREAD

**Issue by huhuhang** [huhuhang/ReadList#189]
**标题**：PyTorch中在反向传播前为什么要手动将梯度清零？

**字数**：414 字

**摘要**：这种模式可以让梯度玩出更多花样，比如说梯度累加（gradient accumulation） 传统的训练函数，一个batch是这么训练的： for ,(iimages,target) in enumerate(train_loader): # 1. input output images = images.cuda(non_blocking=True) target = torch.from_numpy(np.

**链接**：https://www.zhihu.com/question/303070254
